### ISSUE TITLE
Ghost chunks appearing when flying through a pre-loaded world forcing chunks to load.

### ISSUE THREAD

**Issue by mibby** [Kugge/BestViewDistance#25]
BestViewDistance compiled as of commit https://github.com/Kugge/BestViewDistance/commit/8e335e8d61ba77dd4e1c83dfda28e23ac98771f9 with all 5 active pull requests as of 4/4.
Paper dev (Spigot 1.13.2)
ProtocolSupport dev 374

Config
https://paste.ubuntu.com/p/kvhNdtyZ3c/

Flying around in a pre-generate world with ``/vdist`` showing I have a full active 16 chunk render distance, I seem to occasionally come across chunk rendering corruptions.

https://imgur.com/a/380YROc

Flying near / through it and rotating my camera around, the corruption flickers. Sometimes the ghost chunk is in large areas, other times it's just a single 1x1 chunk in the middle of nowhere. Teleporting away and back again, the chunks re-send and display perfectly fine. 

Many players in my community are also reporting a noticeable occurrence of these ghost chunks since BVD was implemented.

@graywolf336 Do you perhaps have any ideas in how to lessen the impact of these ghost chunks?

**Comment by graywolf336** [comment]
Nope, and it happens on my server. I think it is the poor Minecraft client getting extremely confused 😂 I don't have all the time to investigate into why, but if I manage to get some then I'll look into it

**Comment by mibby** [comment]
@graywolf336 Is there any possible way to validate chunks before sending? Or perhaps send slower to prevent issues with the client getting confused? I still seem to run into this issue quite often with dev 18 (your fork) and lots of players asking about it. :(

![link](https://i.imgur.com/Hht0un9.jpg)

**Comment by graywolf336** [comment]
Without getting into packets and intercepting them (which I honestly have no desire to do), I don't know. Can it be reproduced without the plugin and a high static view distance? If so, then that would be an issue with Minecraft or Spigot or Paper that is out of the plugin's control. 🤔

**Comment by mibby** [comment]
I'll have to give it a test in a clean environment later tonight. My production environment is quite heavy.

**Comment by sofianedjerbi** [comment]
Please try without BVD, then with the latest dev version of BVD. Let us know what happens
